bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/NikoHomeControlCommunication.java

```diff
@@ -57,21 +69,69 @@ protected NikoHomeControlCommunication(NhcControllerEvent handler) {
     /**
      * Stop Communication with Niko Home Control system.
      */
-    public abstract void stopCommunication();
+    public void stopCommunication() {
+        stopScheduledRestart();
+
+        resetCommunication();
+    }
+
+    /**
+     * Stop Communication with Niko Home Control system, but keep reconnection attempts going.
+     */
+    public abstract void resetCommunication();
+
+    protected void stopScheduledRestart() {
+        ScheduledFuture<?> future = scheduledRestart;
+        if (future != null) {
+            future.cancel(false);
```

Comment:
In the normal retry cycle, I think so, unless I misunderstand things. stopScheduledRestart is called from checkAndRestartCommunication and is the exit condition there. stopScheduledRestart resets the retry fields, and that would not happen if cancelled with true.
There is another call on stopScheduledRestart from stopCommunication, independent from the retry cycle. There is indeed a small chance there that this will still create a scheduled retry if it is about to reschedule. I should probably synchronize the checkAndRestartCommunication and stopScheduledRestart methods to avoid this.

Reply:
This is not fully clear for me as your code looks relatively complex.
I see also the method `scheduleRestartCommunication` that schedules `scheduleRestart`. Should it be szynchronized too ?

Reply:
@mherwege : gentle ping, especially if you expect a review of your other PR (last part of your old big PR) before the features for the v3.3 are frozen (in 7 days now). For the current PR, I am just expecting an answer to my last question.

Reply:
@lolodomo Sorry for the delay. I wanted to check the flow in my IDE. I don't think it is required, but it won't hurt to synchronize. scheduleRestartCommunication will only do something if scheduledRestartRunnable == null. stopScheduledRestartRunnable sets it to null after cancelling the previous one. If they run at the same time, checkAndRestartCommunication is the one that matters. It is defintely safer to have it synchronized if anything would change in this code. So I have put the synchronization in.

Reply:
Ok, thank you.